packages/components/src/sandbox/index.native.js

```diff
@@ -92,22 +92,62 @@ const style = `
 	body > div iframe {
 		width: 100%;
 	}
-	html.wp-has-aspect-ratio,
-	body.wp-has-aspect-ratio,
-	body.wp-has-aspect-ratio > div,
-	body.wp-has-aspect-ratio > div iframe {
-		height: auto;
-		overflow: hidden; /* If it has an aspect ratio, it shouldn't scroll. */
-	}
 	body > div > * {
 		margin-top: 0 !important; /* Has to have !important to override inline styles. */
 		margin-bottom: 0 !important;
 	}
+
+	.wp-block-embed__wrapper {
+		position: relative;
+	}
+
+	body.wp-has-aspect-ratio > div iframe {
+		height: 100%;
+		overflow: hidden; /* If it has an aspect ratio, it shouldn't scroll. */
+	}
+
+	/* Add responsiveness to embeds with aspect ratios. */
+	.wp-has-aspect-ratio.wp-block-embed__wrapper::before {
+		content: "";
+		display: block;
+		padding-top: 50%; // Default to 2:1 aspect ratio.
+	}
+	.wp-has-aspect-ratio iframe {
+		position: absolute;
+		top: 0;
+		right: 0;
+		bottom: 0;
+		left: 0;
+		height: 100%;
+		width: 100%;
+	}
+	.wp-embed-aspect-21-9.wp-block-embed__wrapper::before {
+		padding-top: 42.85%; // 9 / 21 * 100
+	}
+	.wp-embed-aspect-18-9.wp-block-embed__wrapper::before {
+		padding-top: 50%; // 9 / 18 * 100
+	}
+	.wp-embed-aspect-16-9.wp-block-embed__wrapper::before {
+		padding-top: 56.25%; // 9 / 16 * 100
+	}
+	.wp-embed-aspect-4-3.wp-block-embed__wrapper::before {
+		padding-top: 75%; // 3 / 4 * 100
+	}
+	.wp-embed-aspect-1-1.wp-block-embed__wrapper::before {
+		padding-top: 100%; // 1 / 1 * 100
+	}
+	.wp-embed-aspect-9-16.wp-block-embed__wrapper::before {
+		padding-top: 177.77%; // 16 / 9 * 100
+	}
+	.wp-embed-aspect-1-2.wp-block-embed__wrapper::before {
+		padding-top: 200%; // 2 / 1 * 100
+	}
```

Comment:
In the web version, these styles are applied in a wrapper element to the sandbox component.
https://github.com/WordPress/gutenberg/blob/253d9b6e2123a2806b8843562f3f37de186c933a/packages/block-library/src/embed/embed-preview.js#L114-L118
https://github.com/WordPress/gutenberg/blob/253d9b6e2123a2806b8843562f3f37de186c933a/packages/block-library/src/embed/embed-preview.js#L95

In the native version, we can't do it the same way because the content is being rendered within a WebView, so for this case, we need to include them in the static HTML.

**Styles in the web version:**
https://github.com/WordPress/gutenberg/blob/253d9b6e2123a2806b8843562f3f37de186c933a/packages/block-library/src/embed/style.scss#L38-L89

Reply:
Could there be a way to import the related parts from `embed/style.scss` to JS code here, instead of duplicating? If not, maybe we can add a comment mentioning that it is a copy of the styles in `embed/style.scss`?

Reply:
Not sure if there would be a way, if so I think it could be complex because the styles are converted to the React Native format, which is different from the web so we would need to convert it back to CSS to be used here.

I don't like the idea to have different copies of the same code for both versions but for this case, since the embed previews are rendered in a different way due to the fact that on the native version they're rendered within WebViews, unfortunately, we'll have to maintain a copy.

For now, I'll add a comment to keep a reference to the original styles code 👍 .